src/core/thread/mle.cpp

```diff
@@ -325,13 +325,22 @@ Error Mle::Restore(void)
 
     SuccessOrExit(error = Get<Settings>().Read(networkInfo));
 
+#if OPENTHREAD_MTD
+    VerifyOrExit(!IsActiveRouter(networkInfo.GetRloc16()), error = kErrorInvalidState);
```

Comment:
Not sure if we should still restore MLE frame counter and key sequence even if the device was a router and later became a  MTD.
@abtink @jwhui thoughts?

Reply:
I agree. I think we may need to restore the counters.

Reply:
Yes, we need to restore the frame counters. Reusing frame counters will compromise AES security.